src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -1015,6 +1027,7 @@ private void MoveToMain(Block block, BatchWrite batch, bool wasProcessed)
 
             level.HasBlockOnMainChain = true;
             _chainLevelInfoRepository.PersistLevel(block.Number, level, batch);
+            _bloomStorage.Store(block.Number, block.Bloom);
```

Comment:
reorganizations are not handled - whenever MoveToMain is executed and there are some previous blocks then you have to recalculate all the blooms for the last x blocks where the blocks include higher levels too

however - since reorgs can only add additional bits to blooms then we can actually ignore reorganizations for now and keep all the bloom values

Reply:
Yes, parity ignores reorgs too. This is a tradeof, will add a TODO item, but we can ignore for now.